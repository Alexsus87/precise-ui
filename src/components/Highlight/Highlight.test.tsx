import * as React from 'react';
import { shallow } from 'enzyme';
import { Highlight } from './';

describe('<Highlight />', () => {
  it('should render <Highlight> with background color on search term', () => {
    const wrapper = shallow(<Highlight text="Hello World!" highlight="World" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Highlight> with no matched search term', () => {
    const wrapper = shallow(<Highlight text="Hello World!" highlight="World2" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Highlight> with exact match on case', () => {
    const wrapper = shallow(<Highlight text="Hello World!" highlight="World" ignoreCase={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Highlight> with no exact match on case', () => {
    const wrapper = shallow(<Highlight text="Hello World!" highlight="world" ignoreCase={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Highlight> treating regex patterns as text', () => {
    const wrapper = shallow(<Highlight text="Hello (?)World!" highlight="(?)" ignoreCase={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
