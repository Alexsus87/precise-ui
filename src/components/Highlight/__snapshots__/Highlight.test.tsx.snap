// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render <Highlight> treating regex patterns as text 1`] = `
<span>
  <span
    key="0"
  >
    Hello 
  </span>
  <styled.span
    key="1"
  >
    (?)
  </styled.span>
  <span
    key="2"
  >
    World!
  </span>
</span>
`;

exports[`<Highlight /> should render <Highlight> with background color on search term 1`] = `
<Fragment>
  <span
    key="0"
  >
    Hello 
  </span>
  <styled.span
    key="1"
  >
    World
  </styled.span>
  <span
    key="2"
  >
    !
  </span>
</Fragment>
`;

exports[`<Highlight /> should render <Highlight> with exact match on case 1`] = `
<span>
  <span
    key="0"
  >
    Hello 
  </span>
  <styled.span
    key="1"
  >
    World
  </styled.span>
  <span
    key="2"
  >
    !
  </span>
</span>
`;

exports[`<Highlight /> should render <Highlight> with no exact match on case 1`] = `
<span>
  <span
    key="0"
  >
    Hello World!
  </span>
</span>
`;

exports[`<Highlight /> should render <Highlight> with no matched search term 1`] = `
<span>
  <span
    key="0"
  >
    Hello World!
  </span>
</span>
`;
